### PR TITLE
Do not perform further validation on nullable null data

### LINKED ISSUE #12: Nullable check should terminate further assertions for nullable null
At this point if `$this->schema->nullable === true` and `$this->data === null` we should terminate any futher checks since there is no sence to check any keyword on null which is valid if schema is nullable

https://github.com/lezhnev74/openapi-psr7-validator/blob/master/src/Schema/Validator.php#L68

### PR DESCRIPTION
**Pull request by scaytrase**
Fixes #12

### CODE REVIEW

File: phpunit.xml [hunk 1]
```diff
@@ -6,8 +6,7 @@
          convertErrorsToExceptions="true"
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
-         processIsolation="false"
-         stopOnFailure="true">
+         processIsolation="false">
```
Comment (scaytrase) [inline phpunit.xml:9]: It's hard to debug multiple tests with `stopOnFailure=true`. I can revert it if you against this change

File: tests/FromCommunity/Issue3Test.php [hunk 2]
```diff
@@ -5,20 +5,17 @@
...
 
-
-use cebe\openapi\Reader;
 use GuzzleHttp\Psr7\ServerRequest;
 use OpenAPIValidation\PSR7\ServerRequestValidator;
 use PHPUnit\Framework\TestCase;
 use function GuzzleHttp\Psr7\stream_for;
 
-class ScayTraseTest extends TestCase
+class Issue3Test extends TestCase
```
Comment (scaytrase) [inline tests/FromCommunity/Issue3Test.php:15]: I'm pretty proud for having named tests, but these are spawning fast. I suggest rename them according to fixed issue

File: phpunit.xml (hunk 1 above)
Comment (lezhnev74) [reply phpunit.xml:9]: It's okay, let's leave it as you suggested.

File: tests/FromCommunity/Issue3Test.php (hunk 2 above)
Comment (lezhnev74) [reply tests/FromCommunity/Issue3Test.php:15]: Haha, yeah. Good point.

File: tests/FromCommunity/Issue12Test.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        $validator->validate($psrRequest);
+
+        $this->addToAssertionCount(1);
+    }
+
+    public function getNullableTypeExamples(): array
+    {
+        return [
+            'nullable null' => [null],
```
Comment (scaytrase) [inline tests/FromCommunity/Issue12Test.php:75]: if fix is reverted this test should fail

Comment (lezhnev74) [reply tests/FromCommunity/Issue12Test.php:75]: Yeah, that is good. I also add at least two tests for a feature (red and green) to confirm both ways.

**Comment by lezhnev74** [comment]
OK, looks good to me. Merging.

**Pull request merged**